openmdao/solvers/nonlinear/tests/test_nonlinear_runonce.py

```diff
@@ -54,5 +62,72 @@ def test_feature_solver(self):
 
         assert_rel_error(self, prob['f_xy'], 122.0)
 
+
+@unittest.skipUnless(PETScVector, "PETSc is required.")
+class TestNonlinearRunOnceSolverMPI(unittest.TestCase):
+
+    N_PROCS = 2
+
+    @unittest.skipUnless(MPI, "MPI is not active.")
+    def test_reraise_analylsis_error(self):
+
+        class AEComp(ExplicitComponent):
```

Comment:
nitpick here, but you defined the same component and driver classes in in the above test. Why not move the class definitions out of the test methods?

Reply:
Done